contracts/multiply/GoerliDummyExchange.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity >=0.7.6;
+import "../interfaces/IERC20.sol";
+import "../utils/SafeMath.sol";
+import "../utils/SafeERC20.sol";
+import "hardhat/console.sol";
+
+contract GoerliDummyExchange {
+  using SafeMath for uint256;
+  using SafeERC20 for IERC20;
+
+  address DAI_ADDRESS = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+  mapping(address => bool) public WHITELISTED_CALLERS;
+
+  uint8 slippage;
+
+  uint8 public fee = 0;
+  uint256 public feeBase = 10000;
+
+  address public feeBeneficiaryAddress = 0x70997970C51812dc3A010C7d01b50e0d17dc79C8; // second HH address
+
+  event AssetSwap(
+    address indexed assetIn,
+    address indexed assetOut,
+    uint256 amountIn,
+    uint256 amountOut
+  );
+  
+  modifier onlyAuthorized() {
+    require(WHITELISTED_CALLERS[msg.sender], "Exchange / Unauthorized Caller.");
+    _;
+  }
+
+  constructor(address _beneficiary, uint8 _fee, uint8 _slippage, address _dai, address authorisedCaller){
+    feeBeneficiaryAddress = _beneficiary;
+    fee = _fee;
+    slippage = _slippage;
+    DAI_ADDRESS = _dai;
+    WHITELISTED_CALLERS[authorisedCaller] = true;
```

Comment:
We have map of `WHITELISTED_CALLERS` however in constructor we pass only one `authorisedCaller`